Testing/SystemTests/lib/systemtests/systemtesting.py

```diff
@@ -545,7 +545,14 @@ def __init__(self, total_number_of_tests=0, maximum_name_length=0):
         '''Initialize a class instance, e.g. connect to a database'''
         self._total_number_of_tests = total_number_of_tests
         self._maximum_name_length = maximum_name_length
-        pass
+
+    @property
+    def total_number_of_tests(self):
+        return self._total_number_of_tests
+
```

Comment:
Just out of interest, what is the benefit of using the `@property` decorator as opposed to using an undecorated `def get_total_number...` function?

Reply:
The idea of properties is to really distinguish what should be an attribute that describes something about the object from a method that performs some work. In this case `_total_number_of_tests` is just a fixed attribute of the object and the property mechansim allows you to call it as
```
reporter = TextResultReporter()
reporter.total_number_of_tests  # no parentheses
```
The property system also allows you to step in on set/get differently and even ban setting if you wanted.